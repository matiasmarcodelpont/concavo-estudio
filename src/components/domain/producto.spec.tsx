import '@testing-library/jest-dom'
import { render } from '@testing-library/react'

import { Producto } from './producto'

const producto = {
  slug: 'piso-madera',
  name: 'Piso simil Madera',
}

describe('Producto', () => {
  it('matches snapshot', () => {
    const { container } = render(<Producto {...producto} />)
    expect(container).toMatchSnapshot()
  })

  it('should render the producto image with the alt text', () => {
    const component = render(<Producto {...producto} />)
    component.getByAltText(producto.name)
  })

  it('should render the producto name', () => {
    const component = render(<Producto {...producto} />)
    component.getByText(producto.name)
  })

  it('should not render the dimensions if the producto is from concavo', () => {
    const component = render(<Producto {...producto} />)
    component.queryByRole('dimensions')
  })

  it('should render the dimensions if the producto is from concavo', () => {
    const component = render(<Producto {...producto} concavo />)
    component.getByRole('dimensions')
  })
})
