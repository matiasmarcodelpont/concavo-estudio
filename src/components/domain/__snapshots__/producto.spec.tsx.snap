// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Producto matches snapshot 1`] = `
<div>
  <a
    class="flex flex-col gap-2 cursor-pointer text-dark-gray group"
    href="/productos/piso-madera"
  >
    <div
      class="h-[450px] relative overflow-hidden"
    >
      <img
        alt="Piso simil Madera"
        class="group-hover:scale-105 transition-transform duration-1000"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="100vw"
        src="/_next/image?url=%2Fproductos%2Fpiso-madera.jpg&w=3840&q=75"
        srcset="/_next/image?url=%2Fproductos%2Fpiso-madera.jpg&w=640&q=75 640w, /_next/image?url=%2Fproductos%2Fpiso-madera.jpg&w=750&q=75 750w, /_next/image?url=%2Fproductos%2Fpiso-madera.jpg&w=828&q=75 828w, /_next/image?url=%2Fproductos%2Fpiso-madera.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fproductos%2Fpiso-madera.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fproductos%2Fpiso-madera.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fproductos%2Fpiso-madera.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fproductos%2Fpiso-madera.jpg&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
      />
    </div>
    <div
      class="mb-4 group-hover:text-light-gray transition-colors duration-500 space-y-1"
    >
      <p
        class="text-md text-center uppercase"
      >
        Piso simil Madera
      </p>
    </div>
  </a>
</div>
`;
